### PR TITLE
Implement mySky.setDataLink

### PR DESCRIPTION
**Pull request by m-cat**

### CODE REVIEW

**Comment by dghelm** [comment]
tested, registry entries match skyns:// compatable entries.

File: src/mysky/index.ts [hunk 1]
```diff
@@ -315,6 +317,48 @@ export class MySky {
...
+   * Sets entry at the given path to point to the data link. Like setJSON, but it doesn't upload a file.
+   *
+   * @param path - The data path.
+   * @param dataLink - The data link to set at the path.
+   * @param [customOptions] - Additional settings that can optionally be set.
+   * @returns - An empty promise.
+   */
+  async setDataLink(path: string, dataLink: string, customOptions?: CustomSetJSONOptions): Promise<void> {
+    validateString("path", path, "parameter");
+    validateObject("dataLink", dataLink, "parameter");
```
Comment (dghelm) [inline src/mysky/index.ts:330]: Should be validateString?

**Review by dghelm — CHANGES_REQUESTED** [review]

**Comment by dghelm** [comment]
I think we should hold off on this commit until we know what's going on with Skylinks v2.

**Comment by dghelm** [comment]
We'll want to re-run checks probably since now we _don't_ want these registry entries to match `skynet://` compatible entries.

File: src/mysky/index.ts [hunk 2]
```diff
@@ -313,6 +315,48 @@ export class MySky {
...
+    // Get the latest entry.
+    // TODO: Can remove this once we start caching the latest revision.
+    const getEntryOpts = extractOptions(opts, defaultGetEntryOptions);
+    const signedEntry = await this.connector.client.registry.getEntry(publicKey, dataKey, getEntryOpts);
+    const revision = getRevisionFromSignedEntry(signedEntry);
+
+    // Build the registry entry.
+    const entry: RegistryEntry = {
+      dataKey,
+      data: stringToUint8ArrayUtf8(trimUriPrefix(dataLink, uriSkynetPrefix)),
```
Comment (dghelm) [inline src/mysky/index.ts:350]: I thought I saw a helper function for this in another PR, maybe a follow-up.

Comment (m-cat) [reply src/mysky/index.ts:350]: Doesn't ring any bells for me.

**Review by peterjan — APPROVED** [review]

**Pull request merged**